src/osd/PrimaryLogPG.cc

```diff
@@ -11155,7 +11155,7 @@ bool PrimaryLogPG::start_recovery_ops(
     info.last_complete = info.last_update;
   }
 
-  if (num_missing == num_unfound) {
+  if (num_unfound > 0 && num_missing == num_unfound) {
```

Comment:
This is not right. The original condition covers exact two cases:
(1) Primary does not missing anything;
(2) Primary does have some object missing, and all those objects are currently unfound.

In both cases we should recover replicas first.

So you might want to fix the comments below here instead?

Reply:
Thanks for your review.

> (2) Primary does have some object missing, and all those objects are currently unfound.

I might have missed some context.